### PR TITLE
Use execv for launching backend in driver.

### PR DESCRIPTION
**Pull request by Blealtan**
This PR replaces the `system` call to launch the backend compiler in the driver with `fork + execv`. It avoids potential escaping problems.

Also eliminates redundant quotes yielded from pre-defines.

### CODE REVIEW

**Comment by roastduck** [comment]
There various scenarios about the quotes:

- Whether a path option should be quoted passed to `cmake`?
- Whether a path macro should be quoted passed to `gcc`?
- Whether a path string should be quoted passed to `config` from Python, or `Config` from C++?

We'd better clarify what is desired (to quote or not to quote) for each case in corresponding docstrings, and update `docs/guide/build-and-run.md`.

Or we can try to work out a general solution that accepts both quoted and unquoted paths.

**Comment by Blealtan** [comment]
> * Whether a path option should be quoted passed to `cmake`?

The convention is not to (double) quote it. We just follow it as it is currently. As to writing `CMakeLists.txt`, I'll add corresponded comments in it.

> * Whether a path macro should be quoted passed to `gcc`?

No. `gcc` does not expect quoted strings in arguments such as `-I`. The quotes we saw were for the `sh` to correctly split arguments. For example, with the following `test.sh` script:

```sh
echo $1
```

We will see
```sh
$ sh ./test.sh -I"xxx"
-Ixxx
$ sh test.sh -I"xxx x"
-Ixxx x
```

Thus, by using `execv`, we avoid the previous quotes, as well as other potential escaping issues.
I'll add corresponded notes in `driver.cc`

> * Whether a path string should be quoted passed to `ffi/config.cc` from Python, or `Config` from C++?

Docstrings coming soon.

File: docs/guide/build-and-run.md [hunk 1]
```diff
@@ -44,6 +44,8 @@ There are some options to `cmake`:
 
 - `-DFT_WITH_CUDA=ON/OFF`: build with/without CUDA (defaults to `ON`).
 - `-DFT_WITH_MKL=<path/to/mkl/root>`: build with MKL (path to MKL is required, defaults to building without it).
+
+  The path accepts by CMake should be a raw unescaped path; i.e. `-DFT_WITH_MKL="/some path"` is good since the quotes are resolved by the shell but `-DFT_WITH_MKL=\"/some\ path\"` is not.
```
Comment (roastduck) [inline docs/guide/build-and-run.md:48]: This sentence seems to be inside a Markdown list. Is it rendered correctly?

Comment (Blealtan) [reply docs/guide/build-and-run.md:48]: It is. See [https://github.com/roastduck/FreeTensor/blob/67b2e356c7542580c4896c6b6591999b52ecda06/docs/guide/build-and-run.md](here).

**Review by roastduck — APPROVED** [review]

File: docs/guide/build-and-run.md (hunk 1 above)
Comment (roastduck) [reply docs/guide/build-and-run.md:48]: image

No it's not. @Blealtan

**Pull request merged**